mypy/options.py

```diff
@@ -209,12 +214,16 @@ def __init__(self) -> None:
 
     def snapshot(self) -> object:
         """Produce a comparable snapshot of this Option"""
-        d = dict(self.__dict__)
+        # Under mypyc, we don't have a __dict__, so we need to do worse things.
+        d = dict(getattr(self, '__dict__', ()))
+        for k in get_class_descriptors(Options):
+            if hasattr(self, k):
+                d[k] = getattr(self, k)
         del d['per_module_cache']
         return d
 
     def __eq__(self, other: object) -> bool:
-        return self.__class__ == other.__class__ and self.__dict__ == other.__dict__
+        return isinstance(other, Options) and self.snapshot() == other.snapshot()
```

Comment:
This looks like this `__eq__` will become much more expensive. Have you tracked where/how it is used?

Reply:
I remembered from when I was looking into this while doing the original snapshot stuff that the answer was "not often", but I looked more closely and the answer was: in exactly one test. I just removed `__eq__` and made that test use snapshot directly